scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,417 @@
+            if (userConfig.outputFile is string outputFileValue)
+            {
+                File.AppendAllText(outputFileValue, text);
+            }
+        }
+
+        static void Initialize(UserConfig userConfig)
+        {
+            if (userConfig.outputFile is string outputFileValue && File.Exists(outputFileValue))
+            {
+                File.Delete(outputFileValue);
+            }
+            if (!userConfig.srtEnabled)
+            {
+                WriteToConsoleOrFile(String.Format("WEBVTT{0}{0}", Environment.NewLine), userConfig);
+            }
+        }
+
+/*
+Main functions
+*/
+
+        static UserConfig UserConfigFromArgs(string[] args)
+        {
+// Verify argc >= 3 (caption.exe, subscriptionKey, region)
+            if (args.Length < 3)
+            {
+                throw new TooFewArguments();
+            }
+
+            string[]? languageIDLanguages = null;
+            if (GetCmdOption(args, "-l") is string languageIDLanguagesResult)
+            {
+                languageIDLanguages = languageIDLanguagesResult.Split(',');
+            }
+            
+            return new UserConfig(
+                CmdOptionExists(args, "-f"),
+                CmdOptionExists(args, "-m"),
+                languageIDLanguages,
+                GetCmdOption(args, "-i"),
+                GetCmdOption(args, "-o"),
+                GetCmdOption(args, "-p"),
+                CmdOptionExists(args, "-q"),
+                CmdOptionExists(args, "-u"),
+                GetCmdOption(args, "-r"),
+                CmdOptionExists(args, "-s"),
+                CmdOptionExists(args, "-t"),
+                args[args.Length - 2],
+                args[args.Length - 1]
+            );
+        }
+
+        static AudioConfig AudioConfigFromUserConfig(UserConfig userConfig)
+        {
+            if (userConfig.inputFile is string inputFileValue)
+            {
+                return AudioConfig.FromWavFileInput(inputFileValue);
+            }
+            else
+            {
+                return AudioConfig.FromDefaultMicrophoneInput();
+            }
+        }
+
+        static SpeechConfig SpeechConfigFromUserConfig(UserConfig userConfig)
+        {
+            SpeechConfig speechConfig;
+            if (userConfig.languageIDLanguages is not null)
+            {
+                speechConfig = SpeechConfig.FromEndpoint(new Uri(V2EndpointFromRegion(userConfig.region)), userConfig.subscriptionKey);
+            }
+            else
+            {
+                speechConfig = SpeechConfig.FromSubscription(userConfig.subscriptionKey, userConfig.region);
+            }
+
+            if (userConfig.profanityFilterRemoveEnabled)
+            {
+                speechConfig.SetProfanity(ProfanityOption.Removed);
+            }
+            else if (userConfig.profanityFilterMaskEnabled)
+            {
+                speechConfig.SetProfanity(ProfanityOption.Masked);
+            }
```

Comment:
@v-jaswel - I've found that the default behavior / value is `Masked`, so we will need to change something here. One thing you could do is let them set the exact string following the flag. For example `-f "Raw"`,  `-f "Removed"`, or  `-f "Masked"`.